tools/ossf_best_practices/silver_criteria.md

```diff
@@ -0,0 +1,321 @@
+
+_Possible answers: Met/Unmet/NA_
+
+
+## Automated test suite
+
+> An automated test suite MUST be applied on each check-in to a shared repository for at least one branch. This test suite MUST produce a report on test success or failure.
+
+_Possible answers: Met/Unmet_
+
+> The project MUST add regression tests to an automated test suite for at least 50% of the bugs fixed within the last six months.
+
+_Possible answers: Met/Unmet/NA_
+
+> The project MUST have FLOSS automated test suite(s) that provide at least 80% statement coverage if there is at least one FLOSS tool that can measure this criterion in the selected language.
+
+_Possible answers: Met/Unmet/NA_
+
+
+## New functionality testing
+
+> The project MUST have a formal written policy that as major new functionality is added, tests for the new functionality MUST be added to an automated test suite.
+
+_Possible answers: Met/Unmet/NA_
+
+> The project MUST include, in its documented instructions for change proposals, the policy that tests are to be added for major new functionality.
+
+_Possible answers: Met/Unmet/NA_
+
+## Warning flags
+
+> Projects MUST be maximally strict with warnings in the software produced by the project, where practical.
+
+_Possible answers: Met/Unmet/NA_
+
+
+# Security
+
+## Secure development knowledge
+
+> The project MUST implement secure design principles (from "know_secure_design"), where applicable. If the project is not producing software, select "not applicable" (N/A).
+
+_Possible answers: Met/Unmet/NA_
+
+
+## Use basic good cryptographic practices
+
+_Note that some software does not need to use cryptographic mechanisms. If your project produces software that (1) includes, activates, or enables encryption functionality, and (2) might be released from the United States (US) to outside the US or to a non-US-citizen, you may be legally required to take a few extra steps. Typically this just involves sending an email. For more information, see the encryption section of [Understanding Open Source Technology & US Export Controls](https://www.linuxfoundation.org/resources/publications/understanding-us-export-controls-with-os-projects/)._
+
+> The default security mechanisms within the software produced by the project MUST NOT depend on cryptographic algorithms or modes with known serious weaknesses (e.g., the SHA-1 cryptographic hash algorithm or the CBC mode in SSH).
+
+_Possible answers: Met/Unmet/NA_
+
+> The project SHOULD support multiple cryptographic algorithms, so users can quickly switch if one is broken. Common symmetric key algorithms include AES, Twofish, and Serpent. Common cryptographic hash algorithm alternatives include SHA-2 (including SHA-224, SHA-256, SHA-384 AND SHA-512) and SHA-3.
+
+_Possible answers: Met/Unmet/NA_
```

Comment:
```suggestion
**Met**
```
I'm not really sure if it is applicable tho

Reply:
True, this one is tricky. I am quite sure is `met` as we provide a clear API with several crypto algorithms that the user can interchange easily if one of them is compromised.